required.go

```diff
@@ -6,35 +6,62 @@ import (
 	"strings"
 )
 
-// Required Validator is very simple
-type RequiredValidator struct{}
+// Required Validator ensures a field is "present", certain "falsey" values are handled by
+// the AllowEmpty option .
+type Required struct {
+	AllowEmpty bool
+}
 
 // IsValid determines if the given value satisfies a "Required" rule.
-func (r *RequiredValidator) IsValid(value interface{}) (bool, []error) {
-
+func (r *Required) IsValid(value interface{}) (bool, []error) {
```

Comment:
I don't understand why this needs to return a slice of errors - it looks like everywhere you use it, it only returns a single error.

If you really need to return the info from multiple errors, you should still make it satisfy the error interface, e.g. https://godoc.org/github.com/facebookgo/errgroup#MultiError